File: cplus/leetcode/202511/algo0384.cpp
#include <bits/stdc++.h>
using namespace std;

// 数学算法 - 随机算法 - 6.1 随机数
// 暴力
class Solution {
private:
    vector<int> nums;
public:
    Solution(vector<int>& nums) {
        this->nums = nums;
    }
    
    vector<int> reset() {
        return this->nums;
    }
    
    vector<int> shuffle() {
        int n = this->nums.size();
        vector<int> ans(n);
        vector<int> toAssign;
        for (int i=0; i<n; i++) toAssign.emplace_back(i);

        for (int i = 0; i<n; i++) {
            int j = rand() % toAssign.size();
            ans[i] = nums[toAssign[j]];
            toAssign.erase(toAssign.begin() + j);
        }
        return ans;

    }
};

File: python/algo/202511/algo0384.py
from random import shuffle
from typing import List

# 数学算法 - 随机算法 - 6.1 随机数
class Solution:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def reset(self) -> List[int]:
        return self.nums

    def shuffle(self) -> List[int]:
        arr = self.nums.copy()
        shuffle(arr)
        return arr

